src/mainboard/starfive/visionfive1/bt0/start.S

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+/**
+  ******************************************************************************
+  * @file  start.S
+  * @author  StarFive Technology
+  * @version  V1.0
+  * @date  06/25/2020
+  * @brief
+  ******************************************************************************
+  * @copy
+  *
+  * THE PRESENT SOFTWARE WHICH IS FOR GUIDANCE ONLY AIMS AT PROVIDING CUSTOMERS
+  * WITH CODING INFORMATION REGARDING THEIR PRODUCTS IN ORDER FOR THEM TO SAVE
+  * TIME. AS A RESULT, STARFIVE SHALL NOT BE HELD LIABLE FOR ANY
+  * DIRECT, INDIRECT OR CONSEQUENTIAL DAMAGES WITH RESPECT TO ANY CLAIMS ARISING
+  * FROM THE CONTENT OF SUCH SOFTWARE AND/OR THE USE MADE BY CUSTOMERS OF THE
+  * CODING INFORMATION CONTAINED HEREIN IN CONNECTION WITH THEIR PRODUCTS.
+  *
+  * COPYRIGHT 2020 Shanghai StarFive Technology Co., Ltd.
+  */
+
+//#include "encoding.h"
+//#include "platform.h"
+//#include "sys.h"
+	// SIFIVE:
+	# The previous boot stage passes these variables:
+	#   a0: hartid
+	#   a1: ROM FDT
+	# a0 is redundant with the mhartid register. a1 might not be valid on
+	# some hardware configurations, but is always set in QEMU.
+// END SIFIVE
+
+.section ".bootblock.boot", "ax", %progbits
+
+// write to UART3 (debug)
+_debug:
+  li t5, 0x12440000
+  li t4, 0x42
+  sw t4, 0(t5)
+  li t4, 0x41
+  sw t4, 0(t5)
+  li t4, 0x44
+  sw t4, 0(t5)
+  li t4, '\r'
+  sw t4, 0(t5)
+  li t4, '\n'
+  sw t4, 0(t5)
+  j _start
+
+_start:
+	// This has to be four byted aligned so make our own and call trap_entry
+	la t0, aligned_trap_entry
+	csrw mtvec, t0
+
+	// Allocate 4 KiB stack for each hart
+// csrr t0, mhartid
+// slli t0, t0, 12
+// la sp, _bss_end+4096
+// sub sp, sp, t0
+
+	li	 a1, 0 // NONSMP_HART    ; //
+	csrr a0, mhartid;
+	bne  a0, a1, .LbootOtherHart //other hart
+
+	// WHAT TO DO HERE?
```

Comment:
This whole assemble file can be moved to bt0/src/main.rs Rust source code; the following assembly code initializes a minimal environment for actual programming languages, specifically initialize data and bss sections.

Reply:
Yea, for context: This all stems from the previous implementation that we had here in oreboot (`seeed/beaglev`), and in turn was taken from here:
https://github.com/starfive-tech/JH7100_secondBoot/blob/starfive/boot/start.S